src/java.base/share/classes/java/util/Formatter.java

```diff
@@ -4490,7 +4499,23 @@ private Appendable print(Formatter fmt, StringBuilder sb, TemporalAccessor t, ch
                 }
                 case DateTime.ISO_STANDARD_DATE: { // 'F' (%Y-%m-%d)
                     char sep = '-';
-                    print(fmt, sb, t, DateTime.YEAR_4, l).append(sep);
+                    ChronoField yearField;
+                    if (t instanceof ZonedDateTime
+                            || t instanceof OffsetDateTime
+                            || t instanceof LocalDateTime
+                            || t instanceof LocalDate) {
+                        yearField = ChronoField.YEAR;
```

Comment:
This enumeration of type doesn't seem very principled and may not be correct for an arbitrary Temporal.
(But so far, I don't have a better suggestion).

Reply:
Haven't tried, but I think `t.getChronology() instanceof IsoChronology` should work

Reply:
Now I tried and it didn't compile 🙂 It should have been `t.query(TemporalQueries.chronology()) instanceof IsoChronology`

Reply:
JapaneseChronology is an IsoChronology so run the test again (before committing) to make sure.

Reply:
`JapaneseChronology` is not extending `IsoChronology`, and that is the gist of the change I suggested.

Reply:
I added the testcase of IsoChronology and it also passed.

Reply:
The distinction is pretty subtle, I was referring to the `Chronology.isISOBased()` method. (True for JapaneseChronology).
IsoChronology is final, so there are no subclasses.
This does address the compatibility issue and is an improvement over the original proposal.